tests/webSocket-spec.coffee

```diff
@@ -0,0 +1,523 @@
+        spyOn(protocol, 'initializationMessage').and.callThrough()
+        ws.connect()
+
+        ws.socket.addEventListener 'open', () ->
+          expect(ws.send).toHaveBeenCalled()
+          expect(protocol.initializationMessage).toHaveBeenCalledWith(ws.accessToken)
+          ws.disconnect()
+          done()
+
+      it 'should fire "opened" event when connection open', (done) ->
+        ws = new WebSocketTransport(CONFIG.REALLY_DOMAIN, 'ibj88w5ake', options)
+        ws.connect()
+        spyOn(ws, 'emit').and.callThrough()
+        ws.socket.addEventListener 'open', () ->
+          expect(ws.emit).toHaveBeenCalledWith 'opened'
+          ws.disconnect()
+          done()
+
+    describe 'connection error', ->
+      it 'should fire "error" event', (done) ->
+        ws = new WebSocketTransport('wss://WRONG_ID.really.com', 'ibj88w5ake', options)
+        ws.connect()
+        spyOn(ws, 'emit').and.callThrough()
+        ws.socket.addEventListener 'error', () ->
+          expect(ws.emit).toHaveBeenCalledWith 'error'
+          ws.disconnect()
+          done()
+
+    describe 'connection message', ->
+      it 'should fire "message" event', (done) ->
+        ws = new WebSocketTransport(CONFIG.REALLY_DOMAIN, 'ibj88w5ake', options)
+        ws.connect()
+        spyOn(ws, 'emit').and.callThrough()
+        ws.socket.addEventListener 'message', (e) ->
+          data = JSON.parse e.data
+          expect(ws.emit).toHaveBeenCalledWith 'message', data
+          ws.disconnect()
+          done()
+
+      it 'should handle buffered message callbacks if it has a tag', (done) ->
+        ws = new WebSocketTransport(CONFIG.REALLY_DOMAIN, 'ibj88w5ake', options)
+        ws.connect()
+        ws.socket.addEventListener 'open', () ->
+          message = protocol.createMessage('/users')
+          ws.send(message, {})
+        spyOn(ws.callbacksBuffer, 'handle').and.callThrough()
+        ws.socket.addEventListener 'message', (e) ->
+          expect(ws.callbacksBuffer.handle).toHaveBeenCalledWith JSON.parse e.data
+          ws.disconnect()
+          done()
+
+    describe 'connection close', ->
+      it 'should fire "reconnecting" event when reconnect options is true', (done) ->
+        ws = new WebSocketTransport(CONFIG.REALLY_DOMAIN, 'ibj88w5ake', options)
+        ws.connect()
+        spyOn(ws, 'emit').and.callThrough()
+        ws.socket.addEventListener 'close', () ->
+          expect(ws.emit).toHaveBeenCalledWith 'reconnecting'
+          ws.disconnect()
+          done()
+        ws.socket.close()
+
+      it 'should reconnect when reconnect option is true', (done) ->
+        ws = new WebSocketTransport(CONFIG.REALLY_DOMAIN, 'ibj88w5ake', options)
+        ws.connect()
+        spyOn(ws, 'reconnect').and.callThrough()
+        ws.socket.addEventListener 'close', () ->
+          expect(ws.reconnect).toHaveBeenCalled()
+          ws.disconnect()
+          done()
+        ws.socket.close()
+
```

Comment:
it's better in the future if you merged tests like this; to check `reconnect` event is fired and `reconnect` is called